hbase-client/src/main/java/org/apache/hadoop/hbase/shaded/protobuf/ResponseConverter.java

```diff
@@ -158,52 +156,30 @@ public static SingleResponse getResult(final ClientProtos.MutateRequest request,
         // If there is an exception from the server, the exception is set at
         // the RegionActionResult level, which has been handled above.
         if (actions.hasCondition()) {
-          Result result = null;
-          if (actionResult.getResultOrExceptionCount() > 0) {
-            ResultOrException roe = actionResult.getResultOrException(0);
-            if (roe.hasResult()) {
-              Result r = ProtobufUtil.toResult(roe.getResult(), cells);
-              if (!r.isEmpty()) {
-                result = r;
-              }
-            }
-          }
-          responseValue = new CheckAndMutateResult(actionResult.getProcessed(), result);
+          results.add(regionName, index, getCheckAndMutateResult(actionResult, cells));
         } else {
-          responseValue = actionResult.getProcessed() ?
-            ProtobufUtil.EMPTY_RESULT_EXISTS_TRUE :
-            ProtobufUtil.EMPTY_RESULT_EXISTS_FALSE;
+          results.add(regionName, index, getMutateRowResult(actionResult, cells));
         }
-        results.add(regionName, index, responseValue);
         continue;
       }
 
       if (actions.hasCondition()) {
-        Result result = null;
-        if (actionResult.getResultOrExceptionCount() > 0) {
-          ResultOrException roe = actionResult.getResultOrException(0);
-          Result r = ProtobufUtil.toResult(roe.getResult(), cells);
-          if (!r.isEmpty()) {
-            result = r;
-          }
-        }
-        responseValue = new CheckAndMutateResult(actionResult.getProcessed(), result);
-        results.add(regionName, 0, responseValue);
+        results.add(regionName, 0, getCheckAndMutateResult(actionResult, cells));
       } else {
+        if (actionResult.hasProcessed()) {
```

Comment:
Mind explaing a bit about this change here? Seems the logic is changed?

Reply:
The case where actions.hasCondition() is false and actionResult.hasProcessed() is true is the case for a result of mutateRow(). After this change, mutateRow() can return results of Increment/Append operations. That's why we need to add this logic here.